File: bulk/java/Deserialization01.java
// CWE-502: Insecure Deserialization via ObjectInputStream
import java.io.*;
public class Deserialization01 {
    public Object deserialize(byte[] data) throws Exception {
        ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(data));
        return ois.readObject(); // arbitrary class instantiation
    }
}

File: bulk/csharp/Deserialization01.cs
// CWE-502: Insecure BinaryFormatter deserialization
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;
public class Deserialization01 {
    public object Deserialize(byte[] data) {
        var formatter = new BinaryFormatter();
        return formatter.Deserialize(new MemoryStream(data));
    }
}
